Add Logging In Topics to Astro and Software Docs (#888)
astronomer/docs: software/log-in-to-software.md

---
sidebar_label: 'Log in to Astronomer Software'
title: "Log in to Astronomer Software"
id: log-in-to-software
description: Log in to Astronomer Software to access Astronomer Software features and functionality.
---

You can use the Astronomer Software UI and the Astro CLI to view and modify your workspaces, clusters, environment variables, tasks, and users. Your user credentials need to be authenticated before you can use the Astronomer Software UI or the Astro CLI.

## Log in to the Astronomer Software UI

1. Go to `<basedomain>.astronomer.io`.

2. Select one of the authentication methods used by your organization, or enter your email address and password and click **Log in**.

## Log in to the Astro CLI

Use the Astro CLI to [deploy code to Astro](https://docs.astronomer.io/astro/deploy-code) and create and manage Deployments.

> 1. Install and configure the Astro CLI. See [Get started with the Astro CLI](https://docs.astronomer.io/software/install-cli).

Reviewer: Same comment as in Astro

## Log in to the Astronomer Software UI

1. Open a browser and go to https://cloud.astronomer.io/ and create an Astronomer account.

2. Go to `<basedomain>.astronomer.io`.

3. Select one of the authentication methods used by your organization, or enter your email address and password and click **Log in**.

## Log in to the Astro CLI

Use the Astro CLI to [deploy code to Astro](https://docs.astronomer.io/astro/deploy-code) and create and manage Deployments.

1. Install and configure the Astro CLI. See [Get started with the Astro CLI](https://docs.astronomer.io/software/install-cli).

2. Run the following command:

    ```sh
    astro login <basedomain>
    ```
3. Enter your username and password or use an OAuth token for authentication:

    - Press **Enter**.
    - Copy the URL, open a browser, paste the URL in the address bar, and then press **Enter**. If you're not taken immediately to the Astronomer Auth Token page, log in to Astronomer Software, paste the URL in the address bar, and press **Enter**.

>     - Copy the OAuth token, paste it in the command prompt after **oAuth Token**, and then press **Enter**.

Reviewer: Should we remove these details because CLI tells you exactly what you need to do here? We could just say "Follow the CLI's instructions for your authentication method" or something like that

Author: @jwitz you're not explicitly told what to do. There are some instructions, but it's better to have the entire process documented. If we keep making assumptions and removing content, nothing will be left and there will be no point publishing this content.

    :::info

    If you can't enter your password in the command prompt, your organization is using an alternative authentication method. Contact your administrator, or use an OAuth token for authentication.

    :::     

## Access a different base domain

> A base domain or URL (Uniform Resource Locator) is the static element of a website address. For example, when you visit the Astronomer website, the address bar always displays https://www.astronomer.io no matter what page you access on the Astronomer website.

Reviewer: ```suggestion
A base domain or URL (Uniform Resource Locator) is the static element of a website address. For example, when you visit the Astronomer website, the address bar always displays `https://www.astronomer.io` no matter what page you access on the Astronomer website.
```

---
sidebar_label: 'Log in to Astronomer Software'
title: "Log in to Astronomer Software"
id: log-in-to-software
description: Log in to Astronomer Software to access Astronomer Software features and functionality.
---

> You can use the Astronomer Software UI and the Astro CLI to view and modify your workspaces, clusters, environment variables, tasks, and users. Your user credentials need to be authenticated before you can use the Astronomer Software UI or the Astro CLI. User authentication prevents unauthorized access to organizational data and ensures you have the correct permissions to perform specific tasks.

Reviewer: Same thoughts here as in the Astro doc

## Log in to the Astronomer Software UI

In the Astronomer Software UI you can create and manage workspaces, manage users, add and remove environment variables, view metrics, and view logs.

> 1. Go to https://cloud.astronomer.io/ and create an Astronomer account.

Reviewer: Same thoughts here as in the Astro doc

Author: @jwitz see my previous response.

> 2. Open a browser and go to your organization's Astronomer Software base domain.

Reviewer: ```suggestion
2. Open a browser and go to `<basedomain>.astronomer.io`.
```
We only define a base domain later in the doc, should we add a line about it here?

> 3. Select one of the following options to access the Astronomer Software UI:

Reviewer: Same comment as in the Astro doc

Author: @jwitz see my previous response.

    - To authenticate with with your Auth0 account, click **Log in with Auth0**, and then follow the prompts.
    - To authenticate with your Google account, click **Log in with Google**, choose an account, enter your username and password, and then click **Sign In**.
    - To authenticate with your Okta account, click **Log in with Okta**, enter your username and password, and then click **Sign In**. 
    - To authenticate with with your custom Oauth account, click **Log in with Custom Oauth**, and then follow the prompts.
    - To authenticate with with your Azure AD account, click **Log in with Azure AD**, and then follow the prompts.
    - To authenticate with your internal Okta account, click **Log in with InternalOkta**, enter your username and password, and then click **Sign In**. 
    - To authenticate with your GitHub account, click **Log in with GitHub**, and then follow the prompts.
    - Enter your username and password and click **Log in**.

## Log in to the Astro CLI

> In the Astro CLI you can run a local Airflow environment, apply code changes, view logs for all Airflow components, create deployments, and modify environment variables.

Reviewer: Same thoughts here as in the Astro doc

> 1. Install and configure the Astro CLI. See [Get started with the Astro CLI](https://docs.astronomer.io/astro/cli/get-started).

Reviewer: We should still link to the Software version of this doc, given that we still have weirdness between Software and Astro CLI versions/ usage: https://docs.astronomer.io/software/install-cli

Author: Done

2. Run the following command:

    ```sh

>     astro login

Reviewer: ```suggestion
    astro login <basedomain>
```

    ```

> 3. Enter your email address and press **Enter**.

Reviewer: This process is a bit different for Software. If you have access to a Software environment you can try for yourself, but what generally happens is that:
1. astro login
2. The CLI prompts you for either your login credentials or a blank input. If you enter a blank input, you are provided a link for retrieving an Oauth token from your browser
3. You open the provided link to the Software UI in a web browser, log in using the Software UI (if you aren't logged in already), copy an OAuth token from the Software UI and paste that into the CLI. The screen with the Oauth token is only accessible from the link provided by the CLI
4. Enter either your username + password or the token into the CLI.

4. Press **Enter** to connect your account to Astronomer.

>     If this is your first time logging in, the Astronomer Authorize App dialog appears. Click **Accept** to allow Astronomer to access your profile and email and allow offline access.

Reviewer: This doesn't happen on Software

## Access a different base domain

A base domain or URL (Uniform Resource Locator) is the static element of a website address. For example, when you visit the Astronomer website, the address bar always displays https://www.astronomer.io no matter what page you access on the Astronomer website. The base domain or URL allows computers to quickly identify and access resources in other domains.

> Every cluster is assigned a base domain. If your organization has multiple clusters, you can run Astro CLI commands to quickly move from one base domain to another. This can be useful when you're authenticated on one cluster, but you need to perform tasks on another cluster.

Reviewer: ```suggestion
Every cluster is assigned a base domain. If your organization has multiple clusters, you can run Astro CLI commands to quickly move from one base domain to another.
```
Let's not add too much about the benefits of this process, as the main beneficiaries are going to be Astronomer team members. Most customers will only have one domain/ product to worry about, and will not likely need to make use of the command like this.

Reviewer: Right, I guess my thought is that we don't want to confuse Astro / Software users into thinking that this is more essential of a process than it is. 

If we keep the sentence, we need to change "cluster" to "installation" for Software. It might be better to also say "This is primarily used for" vs. "This can be useful for" so that it communicates "Hey, this is here for this purpose, and if you're not using it for that purpose then you probably don't need to be using it at all!"

Author: @jwitz I think the text is fine as is and becomes very awkward when it is changed to _This is primarily used for_. The definition of _useful_ is:

_**able to be used for a practical purpose** or in several ways._

I have changed _clusters_ to _installation_ in the software topic.

You can authenticate to multiple domains from a single base domain. You run the `astro login x` command to authenticate to a base domain, and then run the `astro context switch x` command to define the default base domain on your computer. 

1. In the Astro CLI, run the following command to re-authenticate to the target base domain:

    ```
    astro login
    ```
2. Run the following command to define the default base domain on your computer:

    ```

>     astro context switch <basedomain>

Reviewer: You would probably want to run `astro context list` first to see what contexts are available before switching.

---
sidebar_label: 'Log in to Astronomer Software'
title: "Log in to Astronomer Software"
id: log-in-to-software
description: Log in to Astronomer Software to access Astronomer Software features and functionality.
---

> You can use the Astronomer Software UI and the Astro CLI to view and modify your workspaces, clusters, environment variables, tasks, and users. Your user credentials need to be authenticated before you can use the Astronomer Software UI or the Astro CLI.

Reviewer 2: Port over Astro doc description we landed on? (Replacing w Software -- but same exact idea?

Author: @paolaperaza with all the edits, it's extremely challenging keeping everything synchronized. This edit has been implemented.

> You can use the Astronomer Software UI and the Astro CLI to view and modify your workspaces, deployments, environment variables, tasks, and users. You need to authenticate your user credentials when you're using the Astronomer Software UI or the Astro CLI for development on Astro.

Reviewer: Same (general) comment here as in the Astro doc

## Prerequisites 

- An Astronomer account.

> - The [Astro CLI](cli/get-started.md)

Reviewer: ```suggestion
- The [Astro CLI](cli/get-started.md).
```
My preference would be no punctuation on lists this small, but it's your call.

Author: @jwitz is this preference documented in our style guide? If not, it should be. Also, it's easier for writers to follow a standard when it's implemented consistently. You've been using punctation in lists elsewhere, so it should be applied here.

## Log in to the Astronomer Software UI

1. Go to `<basedomain>.astronomer.io`.

> 2. Select one of the following options to access the Astronomer Software UI:

Reviewer: ```suggestion
2. Log in to the Software UI using one of the authentication methods that has been configured by your organization.
```

Reviewer: Uncapitalizing organization here because it's in the Software context, where organization is not an RBAC entity.

Author: @jwitz again, these capitalization standards are hard to follow and implement consistently.

    
    - Enter your username and password and click **Log in**.
    - To authenticate with with your Auth0 account, click **Log in with Auth0**, and then follow the prompts.
    - To authenticate with your Google account, click **Log in with Google**, choose an account, enter your username and password, and then click **Sign In**.
    - To authenticate with your Okta account, click **Log in with Okta**, enter your username and password, and then click **Sign In**. 
    - To authenticate with with your custom Oauth account, click **Log in with Custom Oauth**, and then follow the prompts.
    - To authenticate with with your Azure AD account, click **Log in with Azure AD**, and then follow the prompts.
    - To authenticate with your internal Okta account, click **Log in with InternalOkta**, enter your username and password, and then click **Sign In**. 
    - To authenticate with your GitHub account, click **Log in with GitHub**, and then follow the prompts.

>     

Reviewer 2: Probably need to add a line that these options are only available if you configure them. Link to: https://docs.astronomer.io/software/integrate-auth-system

Author: @paolaperaza done.

>     - To authenticate with your GitHub account, click **Log in with GitHub**, and then follow the prompts.

Reviewer: I thought I had commented on this before, but for a lot of users only a few of these options will appear based on how the organization has configured IdP. Offering some alternative wording at the start of the step, but at a minimum we should remove this bulleted list entirely.

Author: @jwitz you can argue about this with @paolaperaza . She asked that I add the information again. I'm not removing it a second time! :)

Reviewer: Is this the comment you are talking about? https://github.com/astronomer/docs/pull/888#discussion_r920748984

I don't interpret this as "add the login information back in". I think she's asking for the additional sentence on line 30 of this PR "To integrate an identity provider (IdP) with Astronomer Software, see [Integrate an auth system on Astronomer Software](integrate-auth-system.md)."

This is separate from my comment, which is that this bulleted list should be cut entirely because no user will ever see all these options - especially InternalOkta, which is only for testing purposes.

Author: @jwitz sorry, I'm confused by you statement. The text you have identified already appears in the topic. I don't think we need to add this twice.

**Software:**

![image](https://user-images.githubusercontent.com/104205257/179030028-13e66d7f-b9cd-4408-bfd2-1dd0b429efa1.png)

**Astro:**

![image](https://user-images.githubusercontent.com/104205257/179030262-b11fda94-e2b5-45db-8b04-5bb8cf3fa76f.png)

Perhaps I don't understand the issue? Also, Paola asked that I list the identity providers as they appear in the UI.

## Log in to the Astro CLI

> Use the Astro CLI to [deploy code to Astro](https://docs.astronomer.io/astro/deploy-code) and create and manage Deployments. Astronomer uses refresh tokens to make sure that you don’t need to log in to the Astro CLI every time you run a command.

Reviewer 2: Copy Astro description here?

Also, this is a link to deploy code doc in Astro, not Software

1. In the Astro CLI, run the following command:

    ```sh
    astro login <basedomain>
    ```
2. Enter your username and password or use an OAuth token for authentication:

    - Press **Enter**.
    - Copy the URL in the command prompt, open a browser, paste the URL in the address bar, and then press **Enter**. If you're not taken immediately to the Astronomer Auth Token page, log in to Astronomer Software, paste the URL in the address bar, and press **Enter**.
    - Copy the OAuth token, paste it in the command prompt after **oAuth Token**, and then press **Enter**.

    :::info

    If you can't enter your password in the command prompt, your organization is using an alternative authentication method. Contact your administrator, or use an OAuth token for authentication.

    :::     

## Access a different base domain

> When you need to access Astro and Astronomer Software with the Astro CLI at the same time, you need to authenticate to each product individually by specifying a base domain for each Astronomer installation.

Reviewer 2: ```suggestion
To access multiple installations of Astronomer Software with the Astro CLI at the same time, you need to authenticate to each cluster individually by specifying its base domain. This is also applicable if you are using both Astro and Astronomer Software. 
```

Something more like this for Software?

> When you need to access multiple installations of Astronomer Software with the Astro CLI at the same time or you need to use Astro and Astronomer Software at the same time, you need to authenticate to each cluster individually by specifying its base domain. 

Reviewer: ```suggestion
To access multiple installations of Astronomer Software with the Astro CLI at the same time or use Astro and Astronomer Software at the same time, you need to authenticate to each cluster individually by specifying its base domain. 
```

> For Astro users, the base domain is `cloud.astronomer.io`. For Astronomer Software, every cluster has a base domain that you must authenticate to in order to access it. If your organization has multiple clusters, you can run Astro CLI commands to quickly move from one base domain to another. This can be useful when you need to move from an Astronomer Software installation to Astro and are using the Astro CLI to perform actions on both accounts.

Reviewer 2: ```suggestion
For Astronomer Software, every cluster has a base domain that you must authenticate to in order to access it. If your organization has multiple clusters, you can run Astro CLI commands to quickly move from one base domain to another. This can be useful when you need to move from an Astronomer Software installation to Astro and are using the Astro CLI to perform actions on both accounts.
```
